src/Plugin/Field/FieldType/OgMembershipReferenceItemList.php

```diff
@@ -122,9 +122,9 @@ public function referencedEntities() {
     }
 
     // Entities should be populated from populateGroupsFromMembershipEntities().
-    return array_map(function($item) {
+    return array_filter(array_map(function($item) {
```

Comment:
Ah, because when the widget is created an additional empty item is added (so you get the empty element to fill in on the form). If you try to get the referenced entities after that, like the other groups widget is doing we don't want to return an empty value in the array of entities.